syntax = "proto3";


service Communicator {
  rpc Heartbeat(stream HB) returns (stream HB) {}
  rpc SendToMaster(MainMessage) returns (MainMessage) {}
  rpc RecvFromMaster(MainMessage) returns (MainMessage) {}
}


message SendTime{
  string task_id = 1;
  string method_name = 2;
  float send_time = 3;
}

message HB{
  string agent_name = 1;
  string status = 2;
  repeated SendTime send_timings = 3;
}

message MainMessage {
  string sender_id = 1;
  optional string task_id = 2;
  optional string method_name = 3;
  map<string, bytes> tensor_kwargs = 4;
  map<string, bytes> other_kwargs = 5;
  map<string, bytes> prometheus_metrics = 6;
  optional float get_response_timeout = 7;
}
